File: libs/grpc/clients/websocket.go
package clients

import (
	"fmt"
	"github.com/satont/tsuwari/libs/grpc/generated/websocket"
	"log"

	"github.com/satont/tsuwari/libs/grpc/servers"
	"google.golang.org/grpc"
	"google.golang.org/grpc/balancer/roundrobin"
	"google.golang.org/grpc/credentials/insecure"
)

func NewWebsocket(env string) websocket.WebsocketClient {
	serverAddress := createClientAddr(env, "websocket", servers.WEBSOCKET_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		grpc.WithTransportCredentials(insecure.NewCredentials()),
		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, roundrobin.Name)),
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := websocket.NewWebsocketClient(conn)

	return c
}

File: libs/grpc/clients/websocket.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc';

import { WebsocketClient, WebsocketDefinition } from '../generated/websocket/websocket.js';
import { PORTS } from '../servers/constants.js';
import { CLIENT_OPTIONS, createClientAddr } from './helper.js';

export const createWebsocket = (env: string): WebsocketClient => {
  const channel = createChannel(
    createClientAddr(env, 'websocket', PORTS.WEBSOCKET_SERVER_PORT),
    ChannelCredentials.createInsecure(),
    CLIENT_OPTIONS,
  );

  const client = createClient(WebsocketDefinition, channel);

  return client as any;
};
